openstudiocore/src/cli/main.cpp

```diff
@@ -229,28 +260,177 @@ int main(int argc, char *argv[])
     Init_trans_utf_16_32();
     rb_provide("enc/trans/utf_16_32.so");
 
-    init_openstudio_internal();
+    // in case any further init methods try to require files, init this first
+    Init_EmbeddedScripting();
+
+    // Need embedded_help for requiring files out of the embedded system
+    auto embedded_extensions_string = embedded_files::getFileAsString(":/embedded_help.rb");
+    rubyInterpreter.evalString(embedded_extensions_string);
+
+    // extensions
+
+    // digest seems to be required early on
+    Init_digest();
+    rb_provide("digest");
+    rb_provide("digest.so");
+
+    Init_bigdecimal();
+    rb_provide("bigdecimal");
+    rb_provide("bigdecimal.so");
+    
+    Init_bubblebabble();
+    rb_provide("bubblebabble");    
+    rb_provide("bubblebabble.so");  
+
+    //Init_callback();
+    //rb_provide("callback");
+    
+    Init_complex();
+    rb_provide("complex");
+    rb_provide("complex.so");
+
+    Init_console();
+    rb_provide("console"); 
+    rb_provide("console.so"); 
+
+    Init_continuation();
+    rb_provide("continuation");
+    rb_provide("continuation.so");
+
+    Init_coverage();
+    rb_provide("coverage");
+    rb_provide("coverage.so");
+
+    Init_cparse();
+    rb_provide("cparse");
+    rb_provide("cparse.so");
+
+    Init_date_core();
+    rb_provide("date_core");
+    rb_provide("date_core.so");
 
     Init_md5();
     rb_provide("digest/md5");
+    rb_provide("digest/md5.so");
+
     Init_rmd160();
     rb_provide("digest/rmd160");
+    rb_provide("digest/rmd160.so");
+
     Init_sha1();
     rb_provide("digest/sha1");
+    rb_provide("digest/sha1.so");
+
     Init_sha2();
     rb_provide("digest/sha2");
+    rb_provide("digest/sha2.so");
+
+    Init_dl();
+    rb_provide("dl");
+    rb_provide("dl.so");
+
+    Init_etc();
+    rb_provide("etc");
+    rb_provide("etc.so");
+
+    Init_fcntl();
+    rb_provide("fcntl");    
+    rb_provide("fcntl.so"); 
+
+    Init_fiber();
+    rb_provide("fiber");
+    rb_provide("fiber.so");
+
+    Init_generator();
+    rb_provide("json/ext/generator");
+    rb_provide("json/ext/generator.so");
+
+    Init_nkf();
+    rb_provide("nkf");
+    rb_provide("nkf.so");
+
+    Init_nonblock();
+    rb_provide("nonblock");
+    rb_provide("nonblock.so");
+
+    Init_objspace();
+    rb_provide("objspace");
+    rb_provide("objspace.so");
+
+    //Init_openssl();
+    //rb_provide("openssl");
+    //rb_provide("openssl.so");
+    
+    Init_parser();
+    rb_provide("json/ext/parser");
+    rb_provide("json/ext/parser.so");
+
+    Init_pathname();
+    rb_provide("pathname");
+    rb_provide("pathname.so");
+    
+    //Init_psych();
+    //rb_provide("psych");
+    //rb_provide("psych.so");
+
+#ifndef _MSC_VER
+    Init_pty();
+    rb_provide("pty");
+    rb_provide("pty.so");
+#endif
+
+    Init_rational();
+    rb_provide("rational");
+    rb_provide("rational.so");
```

Comment:
Kept hitting errors when parts of ruby would require foo.so instead of foo.  I just figured I'd do this for everything just in case to avoid fixing errors one by one as they get exercised.